File: client/src/models/Recipe.js
export class Recipe {

  constructor(data) {
    this.id = data.id;
    this.title = data.title;
    this.instructions = data.instructions;
    this.img = data.img;
    this.category = data.category;
    this.creator_id = data.creator_id;
    this.creator = data.creator;
    this.ingredients = data.ingredients;
    this.favoriteCount = data.favoriteCount;
    this.favorites = data.favorites;
  }
}

File: server/Models/Recipe.cs
using System.Text.Json.Serialization;

namespace allspice.Models;

public class Recipe
{
  [JsonPropertyName("id")]
  public int Id { get; set; }
  public string Title { get; set; }
  public string Instructions { get; set; }
  public string Img { get; set; }
  public string Category { get; set; }
  public string CreatorId { get; set; }
  public Account Creator { get; set; }
  public List<Ingredient> Ingredients { get; set; }

  public List<Favorite> Favorites { get; set; }
  public int FavoriteCount { get; set; }
}
